src/superface-test.utils.ts

```diff
@@ -0,0 +1,243 @@
+  }
+
+  return profile;
+}
+
+/**
+ * Returns provider id if entered provider is either instance of Provider or string
+ */
+export function getProviderName(provider: Provider | string): string {
+  if (provider instanceof Provider) {
+    return provider.configuration.name;
+  }
+
+  return provider;
+}
+
+/**
+ * Returns usecase name if entered usecase is either instance of UseCase or string
+ */
+export function getUseCaseName(useCase: UseCase | string): string {
+  if (useCase instanceof UseCase) {
+    return useCase.name;
+  }
+
+  return useCase;
+}
+
+/**
+ * Returns SuperJson based on path detected with its abstract method.
+ */
+export async function getSuperJson(): Promise<SuperJson> {
+  const superPath = await SuperJson.detectSuperJson(process.cwd(), 3);
+
+  if (superPath === undefined) {
+    throw new SuperJsonNotFoundError();
+  }
+
+  return (await SuperJson.load(joinPath(superPath, 'super.json'))).unwrap();
+}
+
+const HIDDEN_CREDENTIALS_PLACEHOLDER =
+  'credentials-removed-to-keep-them-secure';
+const AUTH_HEADER_NAME = 'Authorization';
+
+export function assertsRecordingsAreNotStrings(
+  recordings: string[] | RecordingDefinition[]
+): asserts recordings is RecordingDefinition[] {
+  for (const recording of recordings) {
+    if (typeof recording === 'string') {
+      throw new UnexpectedError('recording is a string, not object');
+    }
+  }
+}
+
+export async function removeSensitiveInformation(
+  sfConfig: CompleteSuperfaceTestConfig,
+  recordings: RecordingDefinition[]
+): Promise<void> {
+  const {
+    configuration: { security, baseUrl },
+  } = await sfConfig.client.cacheBoundProfileProvider(
+    sfConfig.profile.configuration,
+    sfConfig.provider.configuration
+  );
+
+  for (const recording of recordings) {
+    for (const scheme of security) {
+      removeCredentialsBasedOnScheme(recording, scheme, baseUrl);
+    }
+  }
+}
+
+export function removeCredentialsBasedOnScheme(
+  recording: RecordingDefinition,
+  scheme: SecurityScheme,
+  baseUrl: string
+): void {
+  if (isApiKeySecurityScheme(scheme)) {
+    if (scheme.in === ApiKeyPlacement.HEADER) {
+      if (recording.reqheaders?.[scheme.name] !== undefined) {
+        recording.reqheaders[scheme.name] = HIDDEN_CREDENTIALS_PLACEHOLDER;
+      }
+    } else if (scheme.in === ApiKeyPlacement.BODY) {
+      if (
+        typeof recording.body === 'object' &&
+        !(recording.body instanceof RegExp) &&
+        !(recording.body instanceof Buffer) &&
+        !Array.isArray(recording.body) &&
+        !('includes' in recording.body) &&
+        recording.body[scheme.name] !== undefined
+      ) {
+        recording.body[scheme.name] = HIDDEN_CREDENTIALS_PLACEHOLDER;
+      }
+
+      // TODO: implement hiding credentials from different types of body
+    } else {
+      const recordingPath = new URL(baseUrl + recording.path);
+
+      if (scheme.in === ApiKeyPlacement.PATH) {
+        const regex = new RegExp(`/${scheme.name}=(.*(?=/))/g`);
```

Comment:
I don't think this will work correctly. The way apikey with PATH placement is substituted is that an url looks like `/api/v1/something/{myApiKey}/resource` and the post-substitution looks like `/api/v1/something/abc124key/resource`. So to correctly find the secret we either need to find the value of the secret (and know the value of the secret) or the original pre-substitution url (which is in the map) to find the correct place of substitution and manually overwrite it.

Reply:
Thanks! I can try to remove it based on variables specified in super.json for now. It's up to them to load them before test run.